docs/contributing/building.rst

```diff
@@ -0,0 +1,221 @@
+
+.. code-block:: bash
+
+   wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
+   bash Miniconda3-latest-Linux-x86_64.sh
+
+You will need to log out and back in again to ensure conda is properly
+configured. Then create and activate the development environment:
+
+.. code-block:: bash
+
+   SRC_DIR=<repo/root>
+   conda config --add channels conda-forge
+   conda env create --name katana-dev --file $SRC_DIR/conda_recipe/environment.yml
+   conda activate katana-dev
+   conda install numactl-devel-cos6-x86_64 # For x86_64 builds
+
+Now, run ``cmake`` to configure your build directory and ``make`` to build Katana.
+
+.. code-block:: bash
+
+   BUILD_DIR=$(pwd)/build
+   mkdir -p $BUILD_DIR
+   cd $BUILD_DIR
+   cmake -S $SRC_DIR -B $BUILD_DIR -DKATANA_LANG_BINDINGS=python
+   make
+
+This will build Katana and place the built libraries and executables in
+``$BUILD_DIR``.
+
+Conda Performance
+^^^^^^^^^^^^^^^^^
+
+Conda is slow to install packages.
+This makes installing a new development take a few minutes.
+More importantly, it makes conda package building very slow (~40 minutes for this repository), because the build process installs at least 7 conda environments.
+This can be mitigated by using `Mamba <https://github.com/mamba-org/mamba#the-fast-cross-platform-package-manager>`_.
+Mamba is a (mostly) drop-in replacement for the ``conda`` command that uses a native dependency solver and reduces installation time by 2x in many cases.
+However, Mamba is not as stable or well tested as Conda and does not have the same level of support.
+
+To use Mamba, install it in your conda environment with ``conda install mamba``.
+Then you can use ``mamba install`` as a drop-in replacement for ``conda install``, and similarly for ``mamba env create`` and ``mamba env update``.
+To use Mamba during conda package builds, install `Boa <https://github.com/mamba-org/boa#the-fast-conda-and-mamba-package-builder>`_ with ``mamba install boa``.
+Then you can use ``conda mambabuild`` (*note:* the top level command is ``conda``, *not* ``mamba``) as a replacement for ``conda build``.
+(We are not using Boa proper as the package builder.)
+
+To get a leaner, Mamba using environment in a fresh install, use `Mambaforge <https://github.com/conda-forge/miniforge#mambaforge>`_.
+It is an installer, similar to miniconda, which installs an environment with conda-forge packages and mamba pre-installed (boa must still be installed separately).
+
+Conan
+-----
+
+After running ``scripts/setup_dev_ubuntu.sh``, run the following commands from
+the project source directory to build the system:
+
+.. code-block:: bash
+
+   conan profile update settings.compiler.libcxx=libstdc++11 default
+
+   BUILD_DIR=$(pwd)/build
+   mkdir -p $BUILD_DIR
+   cd $BUILD_DIR
+   conan install $SRC_DIR/config --build=missing
+   cmake -S $SRC_DIR -B $BUILD_DIR -DCMAKE_TOOLCHAIN_FILE=conan_paths.cmake -DKATANA_LANG_BINDINGS=python
```

Comment:
Going forward, we are going to assume people are going to use the Python language bindings. The build obviously supports other configuration options, but if you want to do something different, it is up to you to know what that means and that it is different from the "standard" build.

Reply:
Then why isn't Python language bindings the default? 

What about the other two options I mentioned above? Are they not required?

Reply:
> Then why isn't Python language bindings the default?

Because this is a PR about the documentation of the current state.

> What about the other two options I mentioned above? Are they not required?

They have never been required.